js_test.go

```diff
@@ -1026,3 +1026,33 @@ func TestJetStreamClusterPlacement(t *testing.T) {
 		t.Fatalf("Unexpected tag: %q", v)
 	}
 }
+
+func TestConsumerReplicasOption(t *testing.T) {
+	s := RunBasicJetStreamServer()
+	defer shutdownJSServerAndRemoveStorage(t, s)
+
+	nc, js := jsClient(t, s)
+	defer nc.Close()
+
+	if _, err := js.AddStream(&StreamConfig{Name: "CONSUMER_REPLICAS_TEST", Subjects: []string{"foo"}}); err != nil {
+		t.Fatalf("Error adding stream: %v", err)
+	}
+
+	cb := func(msg *Msg) {}
+	// Subscribe to the stream with a durable consumer "bar".
+	_, err := js.Subscribe("foo", cb, Durable("bar"), ConsumerReplicas(1))
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+
+	// Get consumer info
+	consInfo, err := js.ConsumerInfo("CONSUMER_REPLICAS_TEST", "bar")
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+
+	// Check if the number of replicas is the same as we provided.
+	if consInfo.Config.Replicas != 1 {
```

Comment:
To be fair, this test does not prove anything since even if ConsumerReplicas() was doing nothing, this test would pass, simply because the stream itself is replicas 1, so the new consumer would inherit this value. So the test should either be moved to `test/` directory where we may have some utilities to create a cluster and in that case the stream should be created with replicas 3 and consumer with 1, and in that case this test here is valid, or, before the js.Subscribe() call, create a NATS subscription on the durable create request and check that the request contains the replicas field. Check test `func TestJetStreamConsumerConfigReplicasAndMemStorage()` to see how I am doing this.

Reply:
@kozlovic I totally agree on this. Will make the changes